### PR TITLE
Sub-app content enhancements in IIS topic

### LINKED ISSUE #6027: Hosting .NET Core app as IIS Sub Application
Hi all,

i'm trying to host an .NET Core web application as a sub application under the IIS Default Website and it is behaving very strangely. What I'm looking to do is something like:

- Default IIS Website
  I- DotNetCoreApp (Converted to Application in IIS)

I have converted the .NET Core app to an application and I have created an AppPool with "No Managed Code" and assigned it to it. It all seems fine but when I browse to my .NET Core app it fails to load some static content. Specifically, it seems to serve some content such as the favicon and robots.txt from the http://localhost/DotNetCoreApp/public/ root as expected, but requests to JS files and stylesheets are sent directly to http://localhost/js/myfile.js and http://localhost/css/mystyle.css and they are served correctly but I would expect them to have the DotNetCoreApp prefix as well.

Finally, i have a JS webapplication that is bundled with WebPack in /public/dist/main.<hash>.js which the Home Controller places into the template but unlike the others, the http://localhost/dist/main.<hash>,js does not get served and the /DotNetCoreApp/ is not auto-prefixed.

Also my application does not realise it is in a virtual web root, any URL links to "/mypage" are requesting http://localhost/mypage isntead of http://localhost/DotNetCoreApp/mypage as I would expect.

My simple webconfig file is:

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <system.webServer>
    <handlers>
      <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModule" resourceType="Unspecified" />
    </handlers>
    <aspNetCore processPath="dotnet" arguments=".\MyDotNetCoreApp.dll" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" forwardWindowsAuthToken="false" />
  </system.webServer>
</configuration>
```
and in my Startup.cs I'm using:

```cs
 public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory factory)
        {
            factory.AddConsole(Configuration.GetSection("Logging"));
            factory.AddDebug();

            app.UseStaticFiles();
            app.UseMvc(routes =>
            {
                routes.MapRoute("default", "{*url}", new { controller = "Home", action = "Index" });
            });
        }

        public static void Main()
        {
            Logs.initialize();
            var cwd = Directory.GetCurrentDirectory();
            var web = Path.GetFileName(cwd) == "server" ? "../public" : "public";

            var host = new WebHostBuilder()
                .UseContentRoot(Directory.GetCurrentDirectory())
                .UseWebRoot(web)
                .UseKestrel()
                .UseIISIntegration()
                .UseStartup<Startup>()
                .Build();

            host.Run();
        }
```

Any help with this would be GREATLY appreciated.. It's doing my head in...

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #6027 

[Internal Review Topic (Sub-app section)](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/iis/index?view=aspnetcore-1.0&branch=pr-en-us-9687#sub-applications)

cc: @terndrup

### CODE REVIEW

File: aspnetcore/host-and-deploy/iis/index.md [hunk 1]
```diff
@@ -467,6 +455,22 @@ When hosting a non-ASP.NET Core sub-app underneath an ASP.NET Core app, explicit
 
 ::: moniker-end
 
+Static asset links within the sub-app should use tilde-slash (`~/`) notation. Tilde-slash notation triggers a [Tag Helper](xref:mvc/views/tag-helpers/intro) to prepend the sub-app's pathbase to the rendered relative link. For a sub-app at `/subapp_path`, an image linked with `src="~/image.png"` is rendered as `src="/subapp_path/image.png"`. The root app's Static File Middleware doesn't process the static file request. The request is processed by the sub-app's Static File Middleware.
+
+If a static asset's `src` attribute is set to a relative URL (for example, `src="/image.png"`), the link is rendered without the sub-app's pathbase. The root app's Static File Middleware attempts to serve the asset from the root app's [webroot](xref:fundamentals/index#web-root-webroot), which results in a *404 - Not Found* response unless the static asset is available from the root app.
```
Comment (Tratcher) [inline aspnetcore/host-and-deploy/iis/index.md]: `/image.png` could more specifically be called an "absolute path" as opposed to a "relative path". "relative URL" is ambiguous in this case.

File: aspnetcore/host-and-deploy/iis/index.md [hunk 2]
```diff
@@ -467,6 +455,22 @@ When hosting a non-ASP.NET Core sub-app underneath an ASP.NET Core app, explicit
...
+
+1. Establish an app pool for the sub-app. Set the **.NET CLR Version** to **No Managed Code**.
+
+1. Add the root site in IIS Manager with the sub-app in a folder under the root site.
+
+1. Right-click the sub-app folder in IIS Manager and select **Convert to Application**.
+
+1. In the **Add Application** dialog, use the **Select** button for the **Application Pool** to assign the app pool that you created for the sub-app. Select **OK**.
+
+In some scenarios, URL path segments must be manipulated to make routing work properly. For more information, see <xref:host-and-deploy/proxy-load-balancer#deal-with-path-base-and-proxies-that-change-the-request-path>.
```
Comment (Tratcher) [inline aspnetcore/host-and-deploy/iis/index.md]: The PathBase should not need to be altered manually in IIS scenarios, that's already built in.
https://github.com/aspnet/AspNetCore/blob/4d57528704d7ec54570f5960b3b9ddca503d5c89/src/IISIntegration/src/Microsoft.AspNetCore.Server.IIS/Core/IISServerSetupFilter.cs#L31

File: aspnetcore/host-and-deploy/iis/index.md [hunk 3]
```diff
@@ -467,6 +455,22 @@ When hosting a non-ASP.NET Core sub-app underneath an ASP.NET Core app, explicit
...
+
+To host an ASP.NET Core app as a sub-app under another ASP.NET Core app:
+
+1. Establish an app pool for the sub-app. Set the **.NET CLR Version** to **No Managed Code**.
+
+1. Add the root site in IIS Manager with the sub-app in a folder under the root site.
+
+1. Right-click the sub-app folder in IIS Manager and select **Convert to Application**.
+
+1. In the **Add Application** dialog, use the **Select** button for the **Application Pool** to assign the app pool that you created for the sub-app. Select **OK**.
```
Comment (jkotalik) [inline aspnetcore/host-and-deploy/iis/index.md]: Consider mentioning that for ANCM In-Process, this subapp needs to be in a separate app pool.

**Comment by guardrex** [comment]
@Tratcher @jkotalik Thank you ... I made the updates that you recommended.

**Pull request merged**